route/src/route_generator_worker.cpp

```diff
@@ -281,54 +310,71 @@ namespace route {
         return false;
     }
 
-    std::vector<tf2::Vector3> RouteGeneratorWorker::load_route_destinations_in_ecef(const std::string& route_id) const
+    std::vector<lanelet::BasicPoint3d> RouteGeneratorWorker::load_route_destinations_in_map_frame(const std::vector<cav_msgs::Position3D>& destinations) const
```

Comment:
This might be me being nit-picky, but can we update the route_id based loading to use this same code path? I imagine it must use the route ID to load the file, and make a list of destinations, then can we call this function to complete the loading? Makes sure we always have consistency in interpretation of these route definitions between the two methods.

Reply:
Fixed. I agree that this is good to ensure consistency. I added a function to load the GPS points from the route file specified by the route ID, and then use the same load_route_destinations_in_map_frame() method for both route types.